src/vkscript/command_parser.cc

```diff
@@ -615,9 +615,7 @@ Result CommandParser::ProcessSSBO() {
     if (!r.IsSuccess())
       return r;
 
-    // Resize the buffer so we'll know the max size
-    if (buf->ElementCount() < values.size())
-      buf->SetElementCount(static_cast<uint32_t>(values.size()));
+    buf->SetDataWithOffset(values, cmd->GetOffset());
```

Comment:
I don't think this is going to do what we want either. The trick with VkScript ssbo data is that it's set late. So, this is going to set the data during parse, but we don't want it set until we see the command. So, if you do:

```
ssbo data 1 2 3
draw
ssbo data 1 4 5
draw
```

with this change both draws would see `4 5` in the data, when only the second should see it.

Do you have an example issue where this arises? Is the problem that you need the full buffer size at all times, even outside the resize commands?

Reply:
While creating the the pipeline layout and bindings, which are created before executing commands, pointers to already created buffers are needed (not like Vulkan that needs just the handles). And to create a Dawn buffer, size has to be known. That means I need to know the max size of a buffer while creating the pipeline.

I thought at each ssbo command Vulkan backend resets the data. How about copying the first few lines of that function to just set the size and not the data?

Reply:
I think that makes sense.